### PR TITLE
Implement station signal control

### LINKED ISSUE #116: Assign Signal to station
Needed for #107.

if set the signal of the station is red until the planned departure of train.
Station (=or Player) should control signal, Signal code don't has to be changed

### PR DESCRIPTION
**Pull request by HaSa1002**
Fixes #116
Supersedes #202

Rebase of #202. Original Description:
> Also did some bugfixes.
> Resolves #116
>
> Editor function of station signal control not implemented, because will be deprecated with upcoming scenario editor

~~I am not sure those bug fixes are necessary and my even cause a new bug. But I have to test the current main branch version first.~~
Fixed the issue with stations.

### CODE REVIEW

File: src/Data/Scripts/Player.gd [hunk 1]
```diff
@@ -1411,21 +1415,46 @@ func get_next_SpeedLimit() -> Spatial:
 
 
 var nextStationNode: Spatial = null
+var next_station_index: float = 0 # Used for handle_station_signal, similar to current_station_index. Will be changed once player left the previous station.
```
Comment (DasCapschen) [inline src/Data/Scripts/Player.gd]: why is this a float, if it is an index?

**Review by DasCapschen — APPROVED** [review]
looks good to me, let's see what Jean says before merging.

File: src/Data/Scripts/Player.gd (hunk 1 above)
Comment (HaSa1002) [reply src/Data/Scripts/Player.gd]: that is a bug...

**Comment by HaSa1002** [comment]
superseded by #278

**Pull request closed without merging**